File: financial-statement-studio/apps/api/app/schemas/equity_statement.py
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel


class EquityMovementLine(BaseModel):
    ledger_account_id: str
    account_code: str
    account_name: str

    opening_balance: Decimal
    direct_increases: Decimal
    direct_decreases: Decimal
    net_direct_movement: Decimal
    recorded_closing_balance: Decimal


class EquityMovementSectionLine(BaseModel):
    ledger_account_id: str
    account_code: str
    account_name: str
    amount: Decimal


class EquityMovementSection(BaseModel):
    key: str
    title: str

    items: list[
        EquityMovementSectionLine
    ]

    total: Decimal


class StatementOfChangesInEquityResponse(
    BaseModel,
):
    financial_report_id: str
    company_id: str
    currency: str

    period_start: date
    period_end: date

    opening_recorded_equity: Decimal

    direct_increases: EquityMovementSection
    direct_decreases: EquityMovementSection

    net_direct_equity_movement: Decimal

    profit_after_tax: Decimal

    recorded_closing_equity: Decimal
    total_closing_equity: Decimal

    equity_accounts: list[
        EquityMovementLine
    ]

    calculated_recorded_closing_equity: Decimal

    equity_reconciliation_difference: Decimal

    is_reconciled: bool

    generated_at: datetime

File: financial-statement-studio/apps/web/src/types/equity-statement.ts
export type EquityMoneyValue =
  | string
  | number;

export type EquityMovementLine = {
  ledger_account_id: string;
  account_code: string;
  account_name: string;

  opening_balance: EquityMoneyValue;
  direct_increases: EquityMoneyValue;
  direct_decreases: EquityMoneyValue;
  net_direct_movement: EquityMoneyValue;
  recorded_closing_balance:
    EquityMoneyValue;
};

export type EquityMovementSectionLine = {
  ledger_account_id: string;
  account_code: string;
  account_name: string;
  amount: EquityMoneyValue;
};

export type EquityMovementSection = {
  key: string;
  title: string;

  items: EquityMovementSectionLine[];

  total: EquityMoneyValue;
};

export type StatementOfChangesInEquity = {
  financial_report_id: string;
  company_id: string;
  currency: string;

  period_start: string;
  period_end: string;

  opening_recorded_equity:
    EquityMoneyValue;

  direct_increases:
    EquityMovementSection;

  direct_decreases:
    EquityMovementSection;

  net_direct_equity_movement:
    EquityMoneyValue;

  profit_after_tax:
    EquityMoneyValue;

  recorded_closing_equity:
    EquityMoneyValue;

  total_closing_equity:
    EquityMoneyValue;

  equity_accounts:
    EquityMovementLine[];

  calculated_recorded_closing_equity:
    EquityMoneyValue;

  equity_reconciliation_difference:
    EquityMoneyValue;

  is_reconciled: boolean;

  generated_at: string;
};